src/mlpack/tests/main_tests/logistic_regression_test.cpp

```diff
@@ -616,14 +616,7 @@ BOOST_AUTO_TEST_CASE(LRDecisionBoundaryTest)
   mlpackMain();
 
   // Get the output after first training.
-  const arma::Row<size_t> &output1 = CLI::GetParam<arma::Row<size_t>>("output");
-
-  // Check that the parameters (parameters1 and parameters2) are not equal which
-  // ensures that decision boundary has some effect on the output.
-  // arma::all function checks that each element of the vector is equal to zero.
-  BOOST_REQUIRE_MESSAGE(arma::all(output1 == 0),
-                        "Parameter(Decision Boudary) has"
-                        "no effect on the output");
+  const arma::Row<size_t> output1 = CLI::GetParam<arma::Row<size_t>>("output");
```

Comment:
@rcurtin do you think we can get `output1` by reference just like `output2` ?

Reply:
No; we must preserve the output here since it will be changed when we call `mlpackMain()` again.

Reply:
Yeah, you're right since we're comparing both ouputs they should be different. Reference will change the output1 to output2 and then there'll be no meaning of this test. I didn't see that....my mistake :(